File: KataFizzBuzz.Jasmine/spec/FizzBuzzSpecs.js
describe("FizzBuzz Translation", function() {
	var fizzbuzz;

	it("should translate multiples of three to 'Fizz'", function() {
		var translation = fizzbuzz.translate(6);
		expect(translation).toEqual("Fizz");
	});

	it("should translate multiples of five to 'Buzz'", function() {
		var translation = fizzbuzz.translate(10);
		expect(translation).toEqual("Buzz");
	});

	it("should translate multiples of three and five to 'FizzBuzz'", function() {
		var translation = fizzbuzz.translate(15);
		expect(translation).toEqual("FizzBuzz");
	});

	it("should translate no multiples of three or five to the number itself", function() {
		var translation = fizzbuzz.translate(7);
		expect(translation).toEqual("7");
	});

	beforeEach(function() {
		fizzbuzz = FizzBuzz;
	});
});

File: KataFizzBuzz.HSpec/FizzBuzzSpecs.hs
import Test.Hspec
import Test.QuickCheck
import FizzBuzz

main :: IO ()
main = hspec $ do
  describe "multiples of 3" $ do
    it "returns FIZZ" $ do
      fizzbuzz 6 `shouldBe` "FIZZ"

  describe "multiples of 5" $ do
    it "returns BUZZ" $ do
      fizzbuzz 10 `shouldBe` "BUZZ"

  describe "multiples of 3 and 5" $ do
    it "returns FIZZBUZZ" $ do
      fizzbuzz 30 `shouldBe` "FIZZBUZZ"
